cmd/burrow/commands/restore.go

```diff
@@ -0,0 +1,72 @@
+package commands
+
+import (
+	"github.com/hyperledger/burrow/core"
+	cli "github.com/jawher/mow.cli"
+)
+
+func Restore(output Output) func(cmd *cli.Cmd) {
+	return func(cmd *cli.Cmd) {
+		genesisOpt := cmd.StringOpt("g genesis", "",
+			"Use the specified genesis JSON file rather than a key in the main config, use - to read from STDIN")
+
+		configOpt := cmd.StringOpt("c config", "", "Use the specified burrow config file")
+
+		filename := cmd.StringArg("FILE", "", "Restore from this dump")
+
+		cmd.Spec = "[--config=<config file>] [--genesis=<genesis json file>] [FILE]"
+
+		configOpts := addConfigOptions(cmd)
+
+		cmd.Action = func() {
+			conf, err := obtainBurrowConfig(*configOpt, *genesisOpt)
+			if err != nil {
+				output.Fatalf("could not obtain config: %v", err)
+			}
+
+			err = configOpts.configure(conf)
+			if err != nil {
+				output.Fatalf("could not update burrow config: %v", err)
+			}
+
+			if conf.ValidatorAddress == nil {
+				output.Fatalf("could not finalise validator address - please provide one in config or via --validator-address")
+			}
+
+			output.Logf("Using validator address: %s", *conf.ValidatorAddress)
+
+			kern, err := core.NewKernel()
+			if err != nil {
+				output.Fatalf("could not create Burrow kernel: %v", err)
+			}
+
+			if err = kern.ExecutorFromConfig(conf.Execution); err != nil {
+				output.Fatalf("could not add execution options: %v", err)
+			}
+
+			if err = kern.LoggerFromConfig(conf.Logging); err != nil {
+				output.Fatalf("could not configure logger: %v", err)
+			}
+
+			if err = kern.KeysFromConfig(conf.Keys); err != nil {
+				output.Fatalf("could not configure keys: %v", err)
+			}
+
+			tmConf := conf.Tendermint.TendermintConfig()
+			if err = kern.LoadDump(conf.GenesisDoc, tmConf.DBDir(), *filename); err != nil {
+				output.Fatalf("could not load state: %v", err)
+			}
+
+			if err = kern.TendermintFromConfig(conf.Tendermint, nil, conf.ValidatorAddress); err != nil {
```

Comment:
I thought we had mentioned that `burrow restore` should not start a chain. So do not load tendermint, do not boot, do not shutdown. Perform the load and exit. Only `burrow start` should load a chain. `burrow restore` should be a one-off run - we dont' want to do it every time a pod rolls over.

Reply:
that makes more sense :+1: